napi/transform/src/transformer.rs

```diff
@@ -137,13 +140,18 @@ pub struct TransformResult {
 pub fn transform(
     filename: String,
     source_text: String,
-    options: Option<TransformBindingOptions>,
+    options: Option<TransformOptions>,
 ) -> TransformResult {
     let sourcemap = options.as_ref().is_some_and(|x| x.sourcemap.unwrap_or_default());
     let mut errors = vec![];
 
-    let source_path = Path::new(&filename);
-    let source_type = SourceType::from_path(source_path).unwrap_or_default();
+    let source_type = SourceType::from_path(&filename).unwrap_or_default();
+    let source_type = match options.as_ref().and_then(|options| options.source_type.as_deref()) {
```

Comment:
cc @Boshen Here maybe need to pass `tsx` Extention.

Reply:
why? what's the usecase?